core/src/components/loading/loading.tsx

```diff
@@ -138,6 +139,19 @@ export class Loading implements ComponentInterface, OverlayInterface {
     }
   }
 
+  /**
+   * An ID corresponding to the trigger element that
+   * causes the loading indicator to open when clicked.
+   */
+  @Prop() trigger: string | undefined;
+  @Watch('trigger')
+  triggerChanged() {
+    const { trigger, el, triggerController } = this;
+    if (trigger) {
+      triggerController.initializeClickListener(el, trigger);
```

Comment:
I know we do not explicitly destroy the listener when unmounting the component in modal/popover, but I think we should. The `click` listener likely will not be GC'd when the overlay is removed because the click listener is on the button, not the overlay. Perhaps we can create in `connectedCallback` and destroy in `disconnectedCallback`?

Here's an example of what I mean: https://codepen.io/liamdebeasi/pen/XWqyaWq

Observe that there is a click listener on `ion-button` that was placed there by the modal. (You can remove the `trigger` prop from the modal to see this listener go away). Call `remove()` on the `ion-modal` element. Then manually run garbage collection by clicking the trashcan button in the "Memory" tab in Chrome dev tools. The click listener on `ion-button` should still be there.

Reply:
Should I create a new bug to track `ion-modal` and `ion-popover`? Do we want to fix those as part of 6.x or include this change only with the 7.x changes? 

Instead of adding the event listener in the `connectedCallback`, I think we should do that in `componentDidLoad`. Otherwise the overlay could try to present from a click event on the trigger element, before the overlay has actually rendered.

Reply:
Separate ticket for 6.x is fine

Reply:
Only concern about `componentDidLoad` is how do you re-attach the listener if the node was removed and then re-added to the DOM?

Reply:
Update: `connectedCallback` is the correct call here, since the overlay can be re-added to the DOM.